### ISSUE TITLE
7-Queens crashes in Debug mode

### ISSUE THREAD

**Issue by SSoelvsten** [SSoelvsten/adiar#329]
@anbclausen and @KentNielsen26 has the `queens_B()` function crash for *N = 7* due to line *62-63* in *<adiar/internal/priority_queue.h>* being triggered in a situation where it should not do so.

**Replicability**

- I can replicate it on commit c274e386823a9d502d2d0ffb209a4f417570e84f .
- I cannot replicate myself on my Fedora 36 machine with the newest commit. The main difference is the addition of 2-level cuts and their much tighter approximations. So, the bug still exists, but we just do not know of a situation where it breaks.

**Ideas for what is going on**

- Their [fork](https://github.com/anbclausen/adiar) does include commit 8c9e18cfc538f1815a146bb1028fa55b104d2425 which should fix something like this happening.
- The most obvious reason seems to be that the `memory_usage` and the `memory_fits` functions do not satisfy the following implication, which I would have thought.
 
  **x < memory_fits(M) -> memory_usage(x) < M**

  In other words *memory_fits* is not the inverse of *memory_usage*.
